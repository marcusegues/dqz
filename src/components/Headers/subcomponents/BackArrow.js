import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import Touchable from 'react-native-platform-touchable';
import { MAIN_RED } from '../../../styles/colors';

const BackArrow = ({ onPress }) => (
  <Touchable onPress={onPress}>
    <Ionicons name="md-arrow-back" size={32} color={MAIN_RED} />
  </Touchable>
);

export default BackArrow;
