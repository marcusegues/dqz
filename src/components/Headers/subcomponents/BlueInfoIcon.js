// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';

export const BlueInfoIcon = () => (
  <Touchable onPress={() => {}}>
    <MaterialCommunityIcons
      name="information"
      size={moderateScale(30)}
      color="#006699"
    />
  </Touchable>
);
