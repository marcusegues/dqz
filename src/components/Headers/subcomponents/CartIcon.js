// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { receiptNotificationBadgeStyle } from '../styles/ReceiptNotificationBadge';
import { MAIN_RED } from '../../../styles/colors';

type CartIconProps = {
  navigation: Navigation,
};

export const CartIcon = ({ navigation }: CartIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'QuestionAnswer' })
    }
  >
    <MaterialIcons
      name="shopping-cart"
      size={moderateScale(30)}
      color={MAIN_RED}
      style={receiptNotificationBadgeStyle.receiptIcon}
    />
  </Touchable>
);
