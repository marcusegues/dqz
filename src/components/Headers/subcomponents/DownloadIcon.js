// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { MaterialIcons } from '@expo/vector-icons';
import { MAIN_RED } from '../../../styles/colors';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';

type DownloadIconProps = {
  navigation: Navigation,
};

export const DownloadIcon = ({ navigation }: DownloadIconProps) => (
  <Touchable
    onPress={() => {
      navigation.goBack();
    }}
  >
    <MaterialIcons name="get-app" size={moderateScale(28)} color={MAIN_RED} />
  </Touchable>
);
