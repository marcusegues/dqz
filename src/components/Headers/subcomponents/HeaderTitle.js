import React from 'react';

import { Text } from 'react-native';

const HeaderTitle = ({ text }) => (
  <Text
    style={{
      fontSize: 18,
      color: '#1A1A1A',
      fontFamily: 'roboto_regular',
    }}
  >
    {text}
  </Text>
);

export default HeaderTitle;
