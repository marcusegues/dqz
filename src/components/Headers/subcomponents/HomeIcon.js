// @flow
import React from 'react';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import Entypo from '@expo/vector-icons/Entypo';
import { moderateScale } from '../../../styles/Scaling';
import type { Navigation } from '../../../types/generalTypes';
import { GREY } from '../../../styles/colors';

type HomeIconProps = {
  navigation: Navigation,
};

export const HomeIcon = ({ navigation }: HomeIconProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({ type: 'NAVIGATE', screen: 'MainMenu' })
    }
  >
    <Entypo name="home" size={moderateScale(27)} color={GREY} />
  </Touchable>
);
