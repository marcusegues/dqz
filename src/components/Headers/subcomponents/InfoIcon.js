import React from 'react';
import Touchable from 'react-native-platform-touchable';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../../styles/Scaling';
import { GREY } from '../../../styles/colors';

const InfoIcon = ({ navigation }) => (
  <Touchable
    onPress={() => {
      navigation.goBack();
    }}
  >
    <MaterialCommunityIcons
      name="information"
      size={moderateScale(23)}
      color={GREY}
    />
  </Touchable>
);

export default InfoIcon;
