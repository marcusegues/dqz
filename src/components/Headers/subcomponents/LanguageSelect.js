// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import type { Language } from '../../../i18n/types/locale';
import type { Navigation } from '../../../types/generalTypes';
import { moderateScale } from '../../../styles/Scaling';

type LanguageSelectProps = {
  language: Language,
  navigation: Navigation,
};

export const LanguageSelect = ({
  language,
  navigation,
}: LanguageSelectProps) => (
  <Touchable
    onPress={() =>
      navigation.dispatch({
        type: 'NAVIGATE',
        screen: 'OnBoarding',
      })
    }
  >
    <Text
      style={{
        fontFamily: 'roboto_regular',
        color: '#757575',
        fontSize: moderateScale(16),
      }}
    >
      {language.toUpperCase()}
    </Text>
  </Touchable>
);
