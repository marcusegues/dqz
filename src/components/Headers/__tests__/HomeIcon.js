// @flow
import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
// $FlowFixMe
import Touchable from 'react-native-platform-touchable';
import { HomeIcon } from '../subcomponents/HomeIcon';

describe('HomeIcon Icon TestSuite', () => {
  test('HomeIcon component renders correctly', () => {
    // $FlowFixMe
    const tree = renderer.create(<HomeIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('we can click it', () => {
    const clickFn = jest.fn();
    // $FlowFixMe
    const component = shallow(<HomeIcon navigation={{ dispatch: clickFn }} />);
    component.find(Touchable).simulate('press');
    expect(clickFn).toBeCalled();
    expect(clickFn).toBeCalledWith({
      type: 'NAVIGATE',
      screen: 'MainMenu',
    });
  });
});
