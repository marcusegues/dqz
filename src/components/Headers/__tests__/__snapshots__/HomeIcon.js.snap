// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeIcon Icon TestSuite HomeIcon component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;

exports[`HomeIcon Icon TestSuite renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlatformTouchable
    onPress={[Function]}
  >
    <Icon
      allowFontScaling={false}
      color="#9B9B9B"
      name="home"
      size={42.42857142857143}
    />
  </PlatformTouchable>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Icon
        allowFontScaling={false}
        color="#9B9B9B"
        name="home"
        size={42.42857142857143}
      />,
      "onPress": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": false,
        "color": "#9B9B9B",
        "name": "home",
        "size": 42.42857142857143,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Icon
          allowFontScaling={false}
          color="#9B9B9B"
          name="home"
          size={42.42857142857143}
        />,
        "onPress": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": false,
          "color": "#9B9B9B",
          "name": "home",
          "size": 42.42857142857143,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
